base/ltfiles.dtx

```diff
@@ -1693,8 +1709,10 @@
 %         {Stop \cs{listfiles} being run twice}
 % \changes{v1.0i}{1994/10/18}
 %         {code moved here from ltclass}
+% \changes{v1.2v}{2023/11/15}{Extend file list information}
 %    \begin{macrocode}
-\def\listfiles{%
+\NewDocumentCommand\listfiles{O{}}{%
+  \SetKeys[__kernel/listfiles]{#1}%
```

Comment:
Do the two booleans need to be initialized in `\listfiles`?

This only makes sense if multiple `\listfiles` with different options were used, which is too rare a real case.

Reply:
Suggested alternative?

Reply:
```suggestion
  \@listfiles@sizesfalse
  \@listfiles@hashesfalse
  \SetKeys[__kernel/listfiles]{#1}%
```
or
```suggestion
  \SetKeys[__kernel/listfiles]{sizes=false,hashes=false,#1}%
```

Reply:
Switches start off false, and these are internal - I really don't think we need to worry about resetting them.